src/main/scala/org/ergoplatform/nodeView/wallet/persistence/WalletDigest.scala

```diff
@@ -18,7 +20,44 @@ import scorex.util.Extensions._
   */
 final case class WalletDigest(height: Int,
                               walletBalance: Long,
-                              walletAssetBalances: mutable.LinkedHashMap[EncodedTokenId, Long])
+                              walletAssetBalances: mutable.LinkedHashMap[EncodedTokenId, Long]) {
+
+  /**
+    * Generate a new wallet digest with contents of a given box included into the digest
+    *
+    * @param box - box to include into the digest
+    * @return updated digest
+    */
+  def putBox(box: ErgoBox): WalletDigest = {
+    val mutMap = walletAssetBalances
+
+    box.additionalTokens.toArray.foreach { tokenRec =>
+      val key = EncodedTokenId @@ scorex.util.encode.Base16.encode(tokenRec._1)
+      val value = mutMap.getOrElse(key, 0L) + tokenRec._2
+      mutMap.updated(key, value)
```

Comment:
Here we mutate the walletAssetBalances instance in the original WalletDigest instance.
Either walletAssetBalances should be cloned first or putBox should not create new instance.
I think it is better to make walletAssetBalances immutable in a first place.

Reply:
Done, with immutable walletAssetBalances